src/sentry/migrations/0384_backfill_installation_ids.py

```diff
@@ -0,0 +1,63 @@
+# Generated by Django 2.2.28 on 2023-03-09 23:03
+
+from django.db import migrations
+
+from sentry.monitors.models import MonitorStatus
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+TERMINAL_STATES = [MonitorStatus.PENDING_DELETION, MonitorStatus.DELETION_IN_PROGRESS]
+DEFAULT_ENVIRONMENT_NAME = "production"
+
+
+def backfill_installation_ids(apps, schema_editor):
+    ServiceHook = apps.get_model("sentry", "ServiceHook")
+
+    # The actor id for all ServiceHooks where a project id is not specified is the installation id,
+    # but other actor ids are user ids.
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        ServiceHook.objects.filter(project_id__isnull=True).values_list(
+            "id",
+            "actor_id",
+        ),
+        result_value_getter=lambda item: item[0],
+    )
```

Comment:
Jfyi, you don't have to use this method if you don't want. It's just for backfills that are huge and might take a long time. You can just do 
```
    queryset = RangeQuerySetWrapperWithProgressBar(ServiceHook.objects.filter(project_id__isnull=True))
    for servicehook in queryset:
         ...
```